streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1558,12 +1607,45 @@ private void processStreamThread(final Consumer<StreamThread> consumer) {
         for (final StreamThread thread : copy) consumer.accept(thread);
     }
 
+    /**
+     * Returns runtime information about the local threads of this {@link KafkaStreams} instance.
+     *
+     * @return the set of {@link org.apache.kafka.streams.processor.ThreadMetadata}.
+     * @deprecated since 3.0 use {@link #threadsMetadata()}
+     */
+    @Deprecated
+    @SuppressWarnings("deprecation")
+    public Set<org.apache.kafka.streams.processor.ThreadMetadata> localThreadsMetadata() {
+        return threadsMetadata().stream().map(threadMetadata -> new org.apache.kafka.streams.processor.ThreadMetadata(
+                threadMetadata.threadName(),
+                threadMetadata.threadState(),
+                threadMetadata.consumerClientId(),
+                threadMetadata.restoreConsumerClientId(),
+                threadMetadata.producerClientIds(),
+                threadMetadata.adminClientId(),
+                threadMetadata.activeTasks().stream().map(taskMetadata -> new org.apache.kafka.streams.processor.TaskMetadata(
+                        taskMetadata.taskId().toString(),
+                        taskMetadata.topicPartitions(),
+                        taskMetadata.committedOffsets(),
+                        taskMetadata.endOffsets(),
+                        taskMetadata.timeCurrentIdlingStarted())
+                ).collect(Collectors.toSet()),
+                threadMetadata.standbyTasks().stream().map(taskMetadata -> new org.apache.kafka.streams.processor.TaskMetadata(
+                        taskMetadata.taskId().toString(),
+                        taskMetadata.topicPartitions(),
+                        taskMetadata.committedOffsets(),
+                        taskMetadata.endOffsets(),
+                        taskMetadata.timeCurrentIdlingStarted())
+                ).collect(Collectors.toSet())))
+                .collect(Collectors.toSet());
+    }
+
     /**
      * Returns runtime information about the local threads of this {@link KafkaStreams} instance.
      *
      * @return the set of {@link ThreadMetadata}.
      */
-    public Set<ThreadMetadata> localThreadsMetadata() {
+    public Set<ThreadMetadata> threadsMetadata() {
```

Comment:
Similarly here, name is maybe not the best, happy to change it if there is a better alternative.

Reply:
I think `localThreadMetadata` sounds better to me (actually even better than the original `localThreadsMetadata` tbh), but I'm fine with `threadsMetadata` too

Reply:
nit: I think `localThreadMetadata` does not sound consistent since metadata of multiple stream threads is returned. What about `metadataForLocalThreads()`?

Reply:
I decided to for for the pattern xxxForxxx to keep consistency among different changes.

It is now `metadataForLocalThreads` but happy to change if anyone has reasons against it.

Reply:
@cadonna `localThreadMetadata` still sounds more correct to me than `localThreadsMetadata`. I really can't explain it other than to say that English is weird, and names/titles like this do not always follow the regular rules of grammar/plurals 🤷‍♀️ 

But actually I like your suggestion `metadataForLocalThreads()` even better than any of them, SGTM